[Swift] Crash in PackConformance::getAssociatedConformance → PackType::get when type-checking variadic generic call with deeply nested associated type pack expansion in presence of ambiguous type name

### Description

**Description**

`swiftc` crashes with a null pointer dereference (SEGV at `0x0000000000000008`) when type-checking a call to a variadic generic function whose return type is a deeply nested associated type pack expansion (`repeat (each T).A.A.A.A`), when the module also contains a duplicate declaration of `S` that makes `S` ambiguous. The ambiguity error causes `struct S: P` to fail its conformance check, and the subsequent pack conformance substitution during effects checking dereferences a null pointer. No special compiler flags are required to reproduce.

**Reproducer**

```swift
protocol P {
  associatedtype A: P
}

struct S: P {
  typealias A = S
}

func f<each T: P>(_: repeat each T) -> (repeat (each T).A.A.A.A) {}

f(S(), S(), S())

public struct S {}
```

**Command**

```
swiftc reproduce.swift
```

**Expected behavior**

The compiler should emit diagnostics about the ambiguous `S` and the conformance failure, then exit gracefully. It should not crash.

**Actual behavior**

```
reproduce.swift:6:17: error: 'S' is ambiguous for type lookup in this context
reproduce.swift:5:8: error: type 'S' does not conform to protocol 'P'

💣 Program crashed: Bad pointer dereference at 0x0000000000000008

While evaluating request TypeCheckPrimaryFileRequest(source_file "reproduce.swift")
error: compile command failed due to signal 11
```

**Call chain**

```
TypeCheckPrimaryFileRequest::evaluate
  → typeCheckTopLevelCodeDecl
    → checkTopLevelEffects
      → classifyApply                              (TypeCheckEffects.cpp)
        → GenericFunctionType::substGenericArgs
          → Type::subst
            → expandPackExpansionShape
              → transformPackExpansionType
                → transformDependentMemberType     (TypeSubstitution.cpp)
                  → InFlightSubstitution::lookupConformance
                    → LookUpConformanceInSubstitutionMap::operator()
                      → SubstitutionMap::lookupConformance
                        → ProtocolConformanceRef::getAssociatedConformance
                          → PackConformance::getAssociatedConformance
                            → PackType::get        ← SEGV at 0x8
```

**Root cause**

When `S` is ambiguous, the conformance of `struct S: P` fails and `S.A` is unresolved. During effects checking of `f(S(), S(), S())`, the compiler substitutes the pack type parameter `each T` with `Pack{S, S, S}` and walks the associated type chain `.A.A.A.A`. `PackConformance::getAssociatedConformance` is called for each `.A` step, but because the conformance is broken due to the ambiguity error, it calls `PackType::get` with an invalid or null `ASTContext`, causing the null pointer dereference. The compiler should guard against performing pack conformance substitutions when conformances are in an error state.

**Environment**

- Compiler: Swift 6.3 (swift-6.3-RELEASE)
- Platform: x86_64 Linux (Ubuntu 24.04.4 LTS)
- Command: `swiftc reproduce.swift` (no special flags required)
- Crash site: `swift/lib/AST/PackConformance.cpp` (`PackConformance::getAssociatedConformance`) → `swift/lib/AST/Type.cpp` (`PackType::get`)

--- a/lib/AST/PackConformance.cpp
+++ b/lib/AST/PackConformance.cpp
@@ -135,6 +135,16 @@ PackConformance *PackConformance::getAssociatedConformance(
     Type assocType, ProtocolDecl *protocol) const {
   SmallVector<Type, 4> packElements;
   SmallVector<ProtocolConformanceRef, 4> packConformances;
+  auto &ctx = Protocol->getASTContext();
+
+  // If any pattern's associated conformance lookup fails (e.g. the underlying
+  // conformance is invalid because of a prior diagnostic), fall back to
+  // ErrorType for the corresponding pack element rather than feeding a null
+  // Type into `PackType::get`.
+  auto getAssocType = [&](ProtocolConformanceRef assocConformance) -> Type {
+    return assocConformance ? assocConformance.getType()
+                            : ErrorType::get(ctx);
+  };
 
   auto conformances = getPatternConformances();
   for (unsigned i : indices(conformances)) {
@@ -145,20 +155,18 @@ PackConformance *PackConformance::getAssociatedConformance(
         conformances[i].getAssociatedConformance(assocType, protocol);
       packConformances.push_back(assocConformancePattern);
 
-      auto assocTypePattern = assocConformancePattern.getType();
       packElements.push_back(PackExpansionType::get(
-          assocTypePattern, packExpansion->getCountType()));
+          getAssocType(assocConformancePattern), packExpansion->getCountType()));
     } else {
       auto assocConformanceScalar =
         conformances[i].getAssociatedConformance(assocType, protocol);
       packConformances.push_back(assocConformanceScalar);
 
-      auto assocTypeScalar = assocConformanceScalar.getType();
-      packElements.push_back(assocTypeScalar);
+      packElements.push_back(getAssocType(assocConformanceScalar));
     }
   }
 
-  auto conformingType = PackType::get(Protocol->getASTContext(), packElements);
+  auto conformingType = PackType::get(ctx, packElements);
   return PackConformance::get(conformingType, protocol, packConformances);
 }
 

--- a/test/Generics/pack_conformance_invalid_associated.swift
+++ b/test/Generics/pack_conformance_invalid_associated.swift
@@ -1,0 +1,26 @@
+// RUN: %target-typecheck-verify-swift
+
+// Regression test for a crash in `PackConformance::getAssociatedConformance`
+// when one of the pattern conformances is invalid (here, because `S: P` fails
+// its conformance check after `S` becomes ambiguous). The recursive
+// associated-conformance lookup returned an invalid conformance whose
+// `getType()` is null, and the null leaked into `PackType::get`.
+// See https://github.com/swiftlang/swift/issues/88434.
+
+protocol P {
+  associatedtype A: P // expected-note{{protocol requires nested type 'A'}}
+}
+
+struct S: P { // expected-error{{type 'S' does not conform to protocol 'P'}}
+              // expected-note@-1{{add stubs for conformance}}
+              // expected-note@-2{{found this candidate}}
+              // expected-note@-3{{'S' previously declared here}}
+  typealias A = S // expected-error{{'S' is ambiguous for type lookup in this context}}
+}
+
+func f<each T: P>(_: repeat each T) -> (repeat (each T).A.A.A.A) {}
+
+f(S(), S(), S()) // expected-warning{{result of call to 'f' is unused}}
+
+public struct S {} // expected-note{{found this candidate}}
+                   // expected-error@-1{{invalid redeclaration of 'S'}}

--- a/validation-test/compiler_crashers_fixed/PackType-get-c4a2e5.swift
+++ b/validation-test/compiler_crashers_fixed/PackType-get-c4a2e5.swift
@@ -1,5 +1,5 @@
 // {"kind":"typecheck","original":"ef834ab9","signature":"swift::PackType::get(swift::ASTContext const&, llvm::ArrayRef<swift::Type>)","signatureAssert":"Assertion failed: (Ptr && \"Cannot dereference a null Type!\"), function operator->","signatureNext":"PackConformance::getAssociatedConformance"}
-// RUN: not --crash %target-swift-frontend -typecheck %s
+// RUN: not %target-swift-frontend -typecheck %s
 protocol a {
   associatedtype b
 }

